pyccel/parser/semantic.py

```diff
@@ -1062,6 +1074,9 @@ def incompatible(i_arg, f_arg):
                         get_final_precision(i_arg) != get_final_precision(f_arg) or
                         i_arg.rank != f_arg.rank)
 
+        flag = True
+        if isinstance(expr, FunctionCall):
+            expr = expr.func_name
```

Comment:
Yes `expr` can be a FunctionDef, we pass it [here](https://github.com/pyccel/pyccel/blob/d9bf2a422a9d87a03759cc6504d55d84f85cebaf/pyccel/parser/semantic.py#L4107)

Reply:
Would:
`self._check_argument_compatibility(function_call, args, function_call, is_elemental, error=True)`
Fix this problem and allow us to avoid the if?

Reply:
no function_call, is just a list of call args

Reply:
Then we should either pass the line/column number explicitly or do:
```suggestion
        if isinstance(expr, FunctionCall):
            func_name = expr.func_name
        else:
            func_name = expr
```
And keep passing `expr` as the symbol to obtain the context clues